src/IF.Lastfm.Core/Api/Helpers/PageResponse.cs

```diff
@@ -21,6 +21,10 @@ public interface IPageResponse<out T> : ILastResponse, IEnumerable<T> where T :
         int TotalPages { get; }
 
         int TotalItems { get; }
+        
+        double? From { get; }
```

Comment:
I don't think `From` and `To` should be properties of `PageResponse` - they're only used for the weekly chart methods, so it's mixing concerns a bit.

Instead, you should create a new class for the weekly chart methods that extends `PageResponse` which has these properties. I would also suggest that `From` and `To` should be `DateTime` objects, for usability.

```c#
public class WeeklyChartResponse<T> : PageResponse<T> {
  DateTime From { get; private set; }
  DateTime To { get; private set; }
}
```

Reply:
If I'm not wrong this becomes a bit complicated. The CreateSuccessResponse/CreateErrorResponse from PageResponse would need to be overridden in order to return a WeeklyChartResponse instead of a PageResponse. Also AddPageInfoFromJToken would need its own implementation to handle setting the To/From attributes. 

Since setting the to/from parameters in the PageResponse is a "nice-to-have-feature" and not really necessary another way would be to just remove them from the code. Or do you think that is too lazy?? :)

Reply:
So I took the simple solution and reverted PageResponse back to the version that is at the master branch (removing the To and From attributes that I added earlier). Also removed the checks on to and from in unit/integration tests. After the changes I notices that integration tests for ScrobbleSIngle and UpdateNowPlaying is failing - both locally and on the Appveyor build.